airflow/models/serialized_dag.py

```diff
@@ -102,7 +102,7 @@ def __init__(self, dag: DAG, processor_subdir: str | None = None):
         dag_data = SerializedDAG.to_dict(dag)
         dag_data_json = json.dumps(dag_data, sort_keys=True).encode("utf-8")
 
-        self.dag_hash = hashlib.md5(dag_data_json).hexdigest()
+        self.dag_hash = hashlib_wrapper.md5(dag_data_json, False).hexdigest()
```

Comment:
```suggestion
        self.dag_hash = hashlib_wrapper.md5(dag_data_json, usedforsecurity=False).hexdigest()
```

Let's name the bool, easier to grok later when looking just at the serialization stuff.